src/brogue/MainMenu.c

```diff
@@ -260,136 +279,336 @@ void initializeMenuFlames(boolean includeTitle,
     do {
         if (isApplicationActive()) {
             // Revert the display.
-            overlayDisplayBuffer(state.rbuf, NULL);
-
-            if (!controlKeyWasDown && controlKeyIsDown()) {
-                strcpy(state.buttons[0].text, customNewGameText);
-                drawButtonsInState(&state);
-                buttonCommands[0] = NG_NEW_GAME_WITH_SEED;
-                controlKeyWasDown = true;
-            } else if (controlKeyWasDown && !controlKeyIsDown()) {
-                strcpy(state.buttons[0].text, newGameText);
-                drawButtonsInState(&state);
-                buttonCommands[0] = NG_NEW_GAME;
-                controlKeyWasDown = false;
-            }
+                overlayDisplayBuffer(mainMenu.rbuf, NULL);
 
             // Update the display.
             updateMenuFlames(colors, colorSources, flames);
             drawMenuFlames(flames, mask);
-            overlayDisplayBuffer(shadowBuf, NULL);
-            overlayDisplayBuffer(state.dbuf, NULL);
+                overlayDisplayBuffer(mainShadowBuf, NULL);
+                overlayDisplayBuffer(mainMenu.dbuf, NULL);
 
+                //Show flyout if selected
+                if (isFlyoutActive()) {
+                    overlayDisplayBuffer(flyoutShadowBuf, NULL);
+                    overlayDisplayBuffer(flyoutMenu.dbuf, NULL);
+                }
             // Pause briefly.
             if (pauseBrogue(MENU_FLAME_UPDATE_DELAY)) {
                 // There was input during the pause! Get the input.
                 nextBrogueEvent(&theEvent, true, false, true);
 
-                // Process the input.
-                button = processButtonInput(&state, NULL, &theEvent);
+                    // quickstart a new game
+                    if (theEvent.param1 == 'n' || theEvent.param1 == 'N') {
+                        rogue.nextGame = NG_NEW_GAME;
+                        break;
             }
 
+                    // Process the flyout menu input as needed
+                    if (isFlyoutActive()) {
+                        flyoutIndex = processButtonInput(&flyoutMenu, NULL, &theEvent);
+                        if (flyoutIndex != -1 && theEvent.eventType == MOUSE_UP || theEvent.eventType == KEYSTROKE) {
+                            rogue.nextGame = flyoutButtons[flyoutIndex].command;
+                        }
+                        if (rogue.nextGame == NG_GAME_MODE) {
+                            chooseGameMode();
+                        }
+                    }
+
+                    // Process the main menu input
+                    mainIndex = processButtonInput(&mainMenu, NULL, &theEvent);
+                    if (theEvent.eventType == MOUSE_UP || theEvent.eventType == KEYSTROKE) {   
+                        if (mainIndex != - 1) {
+                            rogue.nextGame = mainButtons[mainIndex].command;
+                        // Hide the flyout menu on the next pass if the user clicked somewhere random or pressed a random key
```

Comment:
Comment should be below the closing brace?

Reply:
I moved it inside the `else if`...

```
    } else if (flyoutIndex == -1) {
        // Hide the flyout menu on the next pass if the user clicked somewhere random or pressed a random key
        rogue.nextGame = NG_NOTHING;
    }
```